### DISCUSSION TITLE
blog/how-brain-detects-shapes

### DISCUSSION THREAD

**Discussion by giscus** [the-guild-org/the-guild-website#1420] [Docs Discussions]
# blog/how-brain-detects-shapes

This is how our brain detects shapes. And so shall the computer...

https://the-guild.dev/blog/how-brain-detects-shapes

**Comment by onurbilenlacin** [comment]
Hi Eytan, 
Firstly, I would like to thank you to you for the codes you shared on github and explained the algorithm by writing this text. However I probably missing some points because the codes is not working what you added like on giff. When I tried to draw rectangle or square the algorithm detects as "open polygon" as per my poits array. I couldn't fix where the issue is. Also I downloaded the codes from github account and there is no working example on the code. Do you have any suggesstion to me which way should i proceed to solve my issue ?
the example of what I have faced as issue on below video. all the types( I tried to draw) is being "open polygon" nor square neither rectangle. 
https://www.youtube.com/watch?v=7G8CVMXpW4M
